src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -43,57 +53,25 @@ internal sealed class EnumConverter<T> : JsonPrimitiveConverter<T>
         // Since multiple threads can add to the cache, a few more values might be added.
         private const int NameCacheSizeSoftLimit = 64;
 
-        public override bool CanConvert(Type type)
```

Comment:
The converter was incorrectly reporting _every_ enum type as being convertible. This brings it back to the default for `JsonConverter<T>` where `type` needs to equal `T`.